### PR TITLE
add embedded dev challenge

### PR DESCRIPTION
**Pull request by apuhl**
First version of a challenge for embedded developer. The idea is to help us to see logic used, source organization.

### CODE REVIEW

File: challenges/pt-br/embedded-dev-challenge/README.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+Antes de começar, leia os nossos [key values](https://www.keyvalues.com/hash) para entender um pouco sobre o que nós priorizamos no desenvolvimento e faça o seu melhor, pois iremos avaliar o teste como se fosse seu melhor esforço ;)
+
+O teste consiste no desenvolvimento de uma biblioteca C para trabalhar com objetos BER-TLV. A biblioteca deve ser capaz de interpretar objetos BER-TLV existentes em memória, e deve permitir que sua estrutura seja facilmente impressa na tela por um programa de usuário.
+
+Exemplo:
+
+Um programa possui em memória o seguinte objeto TLV:
+
+```
+char tlvObject[] = {0xE1, 0x0B, 0xC1, 0x03, 0x01, 0x02,
```
Comment (heyvito) [inline challenges/pt-br/embedded-dev-challenge/README.md]:
```suggestion
uint8_t tlvObject[] = {0xE1, 0x0B, 0xC1, 0x03, 0x01, 0x02,
```

File: challenges/pt-br/embedded-dev-challenge/README.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  LEN – 2 bytes
+  VAL – 0xAA 0xBB
+```
+
+Para este exercício devem ser implementados:
+
+1. Biblioteca que acabamos de especificar, em linguagem C;
+2. Programa que utiliza a biblioteca implementada e imprima o conteúdo de um TLV na tela, tal qual o exemplo acima.
+
+Para auxiliar no desenvolvimento, você encontra, neste repositório, trecho da especificação técnica que explica como os objetos BER-TLV são codificados.
```
Comment (heyvito) [inline challenges/pt-br/embedded-dev-challenge/README.md]:
```suggestion
Para auxiliar no desenvolvimento, você encontra, neste repositório, trecho da especificação técnica que específica o formato de codificação de objetos BER-TLV.
```

File: challenges/pt-br/embedded-dev-challenge/README.md [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+## Restrições
+
+1. Uso da linguagem C.
+2. Biblioteca deve ser um SO (shared object) Linux
```
Comment (heyvito) [inline challenges/pt-br/embedded-dev-challenge/README.md:45]:
```suggestion
2. Biblioteca deve ser um SO (shared object) Linux
3. Processo de build via CMake ou Makefile
```

File: challenges/pt-br/embedded-dev-challenge/README.md [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+
+## Avaliação
+
+1. Conversaremos sobre estrutura de código, lógica e decisões tomadas
+2. Considere a possibilidade de evolução, trabalho em time na manutenção da biblioteca
```
Comment (heyvito) [inline challenges/pt-br/embedded-dev-challenge/README.md]:
```suggestion
2. Considere a possibilidade de evolução e trabalho em time na manutenção da biblioteca
```

**Review by heyvito — CHANGES_REQUESTED** [review]

**Review by heyvito — APPROVED** [review]

**Review by MarcoWorms — APPROVED** [review]
lgtm :shipit:

File: challenges/pt-br/embedded-dev-challenge/README.md [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+## Restrições
+
+1. Uso da linguagem C
+2. Biblioteca deve ser um SO (shared object) Linux
+3. Processo de build via CMake ou Makefile
+
+
+## Avaliação
+
+1. Conversaremos sobre estrutura de código, lógica e decisões tomadas
```
Comment (ghost) [inline challenges/pt-br/embedded-dev-challenge/README.md]: Critério de avaliação a adicionar: código correto/funciona (?)

**Comment by ghost** [comment]
We could perhaps ask the candidate to write a short API spec for his library?

File: challenges/pt-br/embedded-dev-challenge/README.md (hunk 5 above)
Comment (heyvito) [reply challenges/pt-br/embedded-dev-challenge/README.md]: +1: Documentação

**Comment by heyvito** [comment]
> We could perhaps ask the candidate to write a short API spec for his library?

Wouldn't documentation be enough?

**Pull request merged**